routes/api.js

```diff
@@ -1,15 +1,154 @@
 const router = require('express').Router();
+const crypto = require('crypto');
+const statusCode = require('../helper/status_code');
+const jwt = require('jsonwebtoken');
+const User = require('../model/user');
+const userLocation = require('../model/user_location');
+const Image = require('../model/Image');
 
+const secret = 'secretKey';
+
+const SignUpHandler = function signUp(req, res) {
+  const email = req.body.email || undefined;
+  const username = req.body.username || '';
+  const password = req.body.password || undefined;
+
+  if (email && password) {
+    // check whether the email address is used
+    User.findOne({
+      where: {
+        email,
+      },
+    }).then((user) => {
+      if (user) {
+        res.end(
+          statusCode.ERROR_USER_SIGNUP_ACCOUNT,
+        );
+      } else {
+        const md5 = crypto.createHash('md5');
+        md5.update(password);
+
+        const passwdCode = md5.digest('hex').substr(0, 16);
+
+        User.create({
+          email,
+          username,
+          password: passwdCode,
+        }).then((user) => {
+          res.jsonp(
+            {
+              stauts_code: statusCode.SUCCESS,
+              status_message: statusCode.SUCCESS_MSG,
+              user_info: user.get('to_dict'),
+            },
+          );
+        });
+      }
+    });
+  } else {
+    res.end(
+      statusCode.ERROR_PARAM,
+    );
+  }
+};
+
+const SignInHandler = function signIn(req, res) {
+  const email = req.body.email || undefined;
+  const password = req.body.password || undefined;
+
+  if (email && password) {
+    const md5 = crypto.createHash('md5');
+    md5.update(password);
+    const passwdCode = md5.digest('hex').substr(0, 16);
```

Comment:
Not so good, you should mix the password with some private keys, and don't do the truncation which may trigger unexpected problems.

Reply:
Also, I strongly recommend using Passport.js to deal with this. There are some mature ways by using Passport.js.  More References should be viewed: 
- http://passportjs.org/docs/username-password
- http://blog.robertonodi.me/node-authentication-series-email-and-password/
- https://github.com/madhurjain/passport-local-sequelize